app/models/login.rb

```diff
@@ -33,9 +33,9 @@ class InvalidOAuth2Token < StandardError; end
     has_secure_password
   end
 
-  validates :identification, presence: true
-  validates :oauth2_token, presence: true
-  validates :single_use_oauth2_token, presence: true
+  validates :identification, presence: true, uniqueness: true
+  validates :oauth2_token, presence: true, uniqueness: true
+  validates :single_use_oauth2_token, presence: true, uniqueness: true
```

Comment:
Did you try using the [shoulda matcher](http://www.rubydoc.info/gems/shoulda-matchers/Shoulda%2FMatchers%2FActiveRecord%3Avalidate_uniqueness_of) to test this?